tests/test_project_time.py

```diff
@@ -0,0 +1,161 @@
+    assert (
+        time_df_ts == time_range_ts
+    ), f"Ending timestamp does not match: {time_df_ts} vs. {time_range_ts}"
+
+
+def check_exploded_tempo_time(project_time_dim, load_data):
+    """
+    - DF.show() (and probably all arithmetics) use spark session time
+    - DF.toPandas() has weird behavior (i.e., 2012-11-04 07:00 gets listed as 08:00, DThom is filing a bug report to spark)
+    - DF.collect() converts timestamps to system timezone (which is different than spark session timezone!)
+    - A safe solution is to:
+        1) set session time to UTC (this can be set globally in path_to_pyspark/spark-defaults.conf),
+        2) set system/computer time to UTC (if that's doable, no solution yet, this affects .collect())
+        3) save DF to file and then reload (to bypass toPandas bug)
+    """
+    # extract data for comparison
+    model_time = (
+        pd.Series(np.concatenate(project_time_dim.list_expected_dataset_timestamps()))
+        .rename("timestamp")
+        .to_frame()
+    )
+    project_time = project_time_dim.get_time_dataframe()
+    tempo_time = load_data.select("timestamp").distinct().sort(F.asc("timestamp"))
+
+    # save to file and reload
+    data_dir = Path(__file__).absolute().resolve().parents[0] / "data"
+    data_dir.mkdir(exist_ok=True)
+    project_time_file = data_dir / "temp-test_project_time-project_time.parquet"
+    tempo_time_file = data_dir / "temp-test_project_time-tempo_time.parquet"
+    if project_time_file.exists():
+        shutil.rmtree(project_time_file)
+    if tempo_time_file.exists():
+        shutil.rmtree(tempo_time_file)
+    project_time.write.parquet(str(project_time_file))
+    tempo_time.write.parquet(str(tempo_time_file))
+    project_time = pd.read_parquet(project_time_file)
+    tempo_time = pd.read_parquet(tempo_time_file)
+
+    model_time_tz = model_time["timestamp"].dt.tz
+    project_time["timestamp"] = project_time["timestamp"].dt.tz_localize(model_time_tz)
+    tempo_time["timestamp"] = tempo_time["timestamp"].dt.tz_localize(model_time_tz)
+
+    # Checks
+    n_model = model_time["timestamp"].nunique()
+    n_project = project_time["timestamp"].nunique()
+    n_tempo = tempo_time["timestamp"].nunique()
+
+    time = (
+        model_time.join(
+            project_time.assign(project_timestamp=project_time["timestamp"]).set_index(
+                "timestamp"
+            ),
+            on="timestamp",
+            how="outer",
+        )
+        .join(
+            tempo_time.assign(tempo_timestamp=tempo_time["timestamp"]).set_index("timestamp"),
+            on="timestamp",
+            how="outer",
+        )
+        .reset_index(drop=True)
+    )
+
+    mismatch = time[time.isna().any(axis=1)]
+    assert n_model == 366 * 24, n_model
+    assert (
+        len(mismatch) == 0
+    ), f"Mismatch:\nn_model={n_model}, n_project={n_project}, n_tempo={n_tempo}\n{mismatch}"
+
+    # delete files
+    shutil.rmtree(project_time_file)
```

Comment:
Need to ensure these files get deleted if an exception occurs before this point. try/finally or a fixture